utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala

```diff
@@ -81,6 +69,40 @@ object BestEffortJsonSchemaEncoder {
     }
   }
 
+  private def encodeObject(fields: Map[String, _], parentSchema: ObjectSchema): EncodeOutput = {
+    (fields.keys.toList ++ parentSchema.getPropertySchemas.keySet.asScala.toList).distinct.map { key =>
+      val schema = Option(parentSchema.getPropertySchemas.get(key))
+      val value = fields.get(key)
+      ObjectField(key, value, schema, parentSchema)
+    } //we do collect here because we want to do own pre-validation with human readably message (consider doing encode and at the end schema.validate instead of this)
+    .collect(encodeFieldWithSchema.orElse(encodeFieldWithoutSchema))
+    .sequence.map { values => Json.fromFields(values) }
+  }
+
+  private def encodeFieldWithSchema: PartialFunction[ObjectField, WithError[(String, Json)]] = {
+    case ObjectField(fieldName, Some(null), Some(schema), _) if schema.isNullableSchema =>
+      Valid((fieldName, Json.Null))
+    case ObjectField(fieldName, None, Some(_), parentSchema) if parentSchema.getRequiredProperties.contains(fieldName) =>
+      error(s"Missing property: $fieldName for schema: $parentSchema.")
+    case ObjectField(fieldName, Some(value), Some(schema), _) =>
+      encode(value, schema, Some(fieldName)).map(fieldName -> _)
+  }
+
+  private def encodeFieldWithoutSchema: PartialFunction[ObjectField, WithError[(String, Json)]] = {
+    case PatternPropertySchema(schema, ObjectField(fieldName, Some(value), _, _)) =>
```

Comment:
maybe we should left comment that now if patternProperties overlaps with properties, we chooses schema from property as schema used for encoding?